ens/main.py

```diff
@@ -29,8 +46,30 @@
     raw_name_to_hash,
 )
 
+if TYPE_CHECKING:
+    from web3 import Web3  # noqa: F401
+    from web3.contract import (  # noqa: F401
+        Contract,
+    )
+    from web3.providers import (  # noqa: F401
+        BaseProvider,
+    )
+
+
 ENS_MAINNET_ADDR = '0x314159265dD8dbb310642f98f50C066173C1259b'
 
+TxDict = TypedDict("TxDict", {
+    "nonce": int,
+    "gasPrice": int,
+    "gas": int,
+    "from": Union[ChecksumAddress, str],
+    "to": Union[ChecksumAddress, str],
```

Comment:
Worth verifying that `str` is accurate (allowed).  What would the value be for `str`?  Also, do we allow raw binary addresses?  If so then `Address` should be part of this too....

Reply:
A `str` could be an ens name - in #1497 I've updated `str` to a `ENS` type to be more specific. Also updated to include `Address` since raw binary addresses are accepted